### PR TITLE
Add prometheus exporter for coordinator

### PR DESCRIPTION
**Pull request by VincentFF**
## What do these changes do?

Add a prometheus exporter for coordinator to monitor computing performance.
 
image

## Related issue number

[#288 Monitoring for GraphScope](https://github.com/alibaba/GraphScope/issues/288)

Fixes

### CODE REVIEW

**Review by sighingnow — APPROVED** [review]
LGTM.

**Comment by sighingnow** [comment]
> ## What do these changes do?
> Add a prometheus exporter for coordinator to monitor computing performance.
> 
> image

Looks pretty good!

**Comment by VincentFF** [comment]
Add a log parsing script to monitor GraphScope GIE.    
The GIE monitor UI as follows:  

99E006DA-154E-45a8-8A9D-232A027ADE09

File: k8s/prometheus/__init__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#! /usr/bin/env python3
```
Comment (lidongze0629) [inline k8s/prometheus/__init__.py:1]: 这个文件删掉

File: k8s/prometheus/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+prometheus-client==0.14.1
```
Comment (lidongze0629) [inline k8s/prometheus/requirements.txt:1]: 这个文件删掉，这个依赖写到 README.md 中

File: coordinator/gscoordinator/coordinator.py [hunk 3]
```diff
@@ -1650,6 +1657,26 @@ def parse_sys_args():
...
+        default=True,
+        help="Enable monitor or not.",
+    )
+    parser.add_argument(
+        "--monitor_port",
+        type=int,
+        default=9968,
+        help="Coordinator service port.",
+    )
+    parser.add_argument(
```
Comment (lidongze0629) [inline coordinator/gscoordinator/coordinator.py:1674]: 这个 “monitor_host” 参数应该不需要，类似这里代码中写 0.0.0.0https://github.com/alibaba/GraphScope/blob/main/coordinator/gscoordinator/coordinator.py#L1732

File: coordinator/gscoordinator/monitor.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    17: "TO_DIRECTED",
+    18: "TO_UNDIRECTED",
+    19: "CLEAR_EDGES",
+    20: "CLEAR_GRAPH",
+    21: "VIEW_GRAPH",
+    22: "INDUCE_SUBGRAPH",
+    23: "UNLOAD_CONTEXT",
+    31: "CREATE_INTERACTIVE_QUERY",
+    32: "SUBGRAPH",
+    33: "GREMLIN_QUERYutil",
```
Comment (lidongze0629) [inline coordinator/gscoordinator/monitor.py:59]: GREMLIN_QUERY

File: k8s/prometheus/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+   <br>
+   To specify the host and port of exporter, use `--monitor_port 12345` and `--monitor_host 0.0.0.0`. To disable exporter, use `--monitor False`.  
+   ```bash 
+   $ cd GraphScope/coordinator
+
+   # default linstening on 127.0.0.1:9968
+   $ python3 -m gscoordinator --num_workers 1 --hosts localhost --log_level INFO --timeout_seconds 600 --port 50254 --cluster_type hosts --instance_id svuifn --vineyard_shared_mem 4G
+
+   # specify the host to 0.0.0.0 and port to 12345
+   $ python3 -m gscoordinator --num_workers 1 --hosts localst --log_level INFO --timeout_seconds 600 --port 50254 --cluster_type hosts --instance_id svuifn --vineyard_shared_mem 4G --monitor_port 12345 --monitor_host 0.0.0.0
```
Comment (lidongze0629) [inline k8s/prometheus/README.md:68]: typo: localst

File: coordinator/gscoordinator/coordinator.py [hunk 6]
```diff
@@ -1650,6 +1657,26 @@ def parse_sys_args():
         default="registry.cn-hongkong.aliyuncs.com/graphscope/dataset:{__version__}",
         help="Docker image to mount the dataset bucket",
     )
+    parser.add_argument(
+        "--monitor",
+        type=str2bool,
+        nargs="?",
+        const=True,
+        default=True,
```
Comment (lidongze0629) [inline coordinator/gscoordinator/coordinator.py:1665]: 默认为 false

File: k8s/prometheus/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+       # metrics_path defaults to '/metrics'
+       # scheme defaults to 'http'.
+
+       static_configs:
+         - targets: ["127.0.0.1:9969"]
+   ```
+   
+   See [prometheus documentation](https://prometheus.io/docs/introduction/first_steps/) for more details.  
+
+4. start a grafana server and create a monitor dashboard.
```
Comment (lidongze0629) [inline k8s/prometheus/README.md:47]: 在 Grafana 这里，把 7个 panel 的 Query Code 写在这里，方便用户配置

File: k8s/prometheus/README.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+   $ python3 -m gscoordinator --num_workers 1 --hosts localst --log_level INFO --timeout_seconds 600 --port 50254 --cluster_type hosts --instance_id svuifn --vineyard_shared_mem 4G --monitor_port 12345 --monitor_host 0.0.0.0
+   ```
+2. start a prometheus server and scrape metrics from coordinator exporter.  
+   <br> 
+   Here [coordinator-grafana.json](https://gist.githubusercontent.com/VincentFF/23613b8f833a7751ce39483ce6643b6a/raw/d40f4e22bb460b4ba04331da15d8129170935340/coordinator-grafana.json) is a dashboard example for gie monitor. You can import it on grafana dashboard to easily create a gie monitor dashboard.  
+   This dashboard integrates node metrics. To collect node metrics, see [node-exporter](https://prometheus.io/docs/guides/node-exporter/).  
+   <br/>
+   See [prometheus documentation](https://prometheus.io/docs/introduction/first_steps/) for more details.  
+
+3. start a grafana server and make a monitor dashboard.
```
Comment (lidongze0629) [inline k8s/prometheus/README.md:77]: Coordinator 中对应的 panel 的 Query Code 也都列出来

**Pull request merged**